src/external/VTK.jl

```diff
@@ -0,0 +1,49 @@
+# Uses WriteVTK.jl to convert scfres structure to VTk file format
+import WriteVTK: vtk_grid, vtk_save
+
+"""
+    save_scfres(VTKFileName, scfres)
+
+The function takes in the VTK filename and the scfres structure and stores into a VTK file.
+Grid Values:
+ρ_real -> real values of scfers.ρ
+ψreal_i_j -> Real values of Bloch waves in real space, where i and j are Kpoint and EigenVector indexes respectively
+ρspin -> Real of ρspin are stored if ρspin in present
+MetaData:
+Energy and occupation.
+"""
+function save_scfres(VTKFileName::AbstractString, scfres)
```

Comment:
Be careful here: This function has the identical signature to https://github.com/JuliaMolSim/DFTK.jl/blob/master/src/jld2io.jl#L38, so this will cause trouble. I think what I would do is make this function as well as https://github.com/JuliaMolSim/DFTK.jl/blob/master/src/jld2io.jl#L25 a three-argument function, where the third argument (`format`) is a `Val`-type: [Val](https://docs.julialang.org/en/v1/base/base/#Base.Val) (e.g. `Val(:jld2)` and `Val(:vtk)`)). Then you can make a two-argument version of the function which can then use the file extension to either call into https://github.com/JuliaMolSim/DFTK.jl/blob/master/src/jld2io.jl#L25 or into the function you code here. Of course that generic `save_scfres(jld::JLD2.JLDFile, scfres::NamedTuple)` needs to be always available to DFTK, so needs to be in a file that is always loaded (not with Requires). With a generic three-argument version of `save_scfres` you can catch cases for formats that are not implemented and give users a graceful error message.

Reply:
Sounds good. Although the functions behave correctly when only one of `JLD2` or `WriteVTK` is called up, what you suggest will be a lot cleaner way to do things. 
P.S. - Thanks for the `Val` tip, didn't know such a dispatch hack existed.

Reply:
I took some liberty to use 2 argument `save_scfres` definition in `dummy_definitions.jl` and since it was no longer containing only dummy definitions I renamed it to `save_load_scfres.jl` . I have also added some `MethodError` catches to provide more meaningful error phrases to the user.